disambiguation.py

```diff
@@ -8,78 +8,93 @@
-                    if token.type == 'w' or token.type == 'c':
-                        # CODE FOR POS
-                        for ps in token.gloss.ps:
-                            sent.append((token.token, ps))
-                        # CODE FOR TONE
-                        # sent.append((token.token, token.gloss.form.encode('utf-8')))
-                        # CODE FOR GLOSS
-                        # sent.append((token.token, token.gloss.gloss.encode('utf-8')))
-                    if token.type == 'c' and token.token in ['.', '?', '!']:
-                        if len(sent) > 1:
-                            allsents.append(sent)
-                        sent = []
-
-        print 'Split the data in train / evaluation'
-        pourcentage = 10;
-        datalength = len(allsents)
-        train_len = (datalength*pourcentage)/100
-        train_set = allsents[:train_len]
-
-        print 'Building CRF classifier (NLTK)'
-        tagger = CRFTagger(verbose = True, training_opt = {'feature.minfreq' : 10})
-        t1 = time.time()
-        tagger.train(train_set, 'disambiguation.model')
-        t2 = time.time()
-        texec = t2-t1
-        print "... done in",  time.strftime('%H %M %S', time.localtime(texec))
-
-    exit(0)
+	
+	aparser = argparse.ArgumentParser(description='Daba disambiguator')
+	# aparser.add_argument('-i', '--infile', help='Input file (.html)', default="sys.stdin")
+	# aparser.add_argument('-o', '--outfile', help='Output file (.html)', default="sys.stdout")
+	aparser.add_argument('-l', '--learn', help='Learn model from data (and save as F if provided)', default=None)
+	aparser.add_argument('-p', '--pos', help='Prediction for POS', default=False, action='store_true')
+	aparser.add_argument('-t', '--tone', help='Prediction for tones', default=False, action='store_true')
+	aparser.add_argument('-g', '--gloss', help='Prediction for gloses', default=False, action='store_true')
+	aparser.add_argument('-e', '--evalsize', help='Percent of randomized data to use for evaluation (default 10)', default=10)
+	aparser.add_argument('-v', '--verbose', help='Verbose output', default=False, action='store_true')
+	args = aparser.parse_args()
+
+	if args.learn:
+
+		if not args.pos or args.tone or args.gloss:
+			print 'Choose pos, tone, gloss or combination of them'
+			exit(0)
+
+		print 'Make list of files'
+		files1 = glob.iglob("../corbama/*/*.dis.html")
+		files2 = glob.iglob("../corbama/*.dis.html")
+		allfiles = ""
+		for file1, file2 in zip(files1, files2):
+			allfiles += file1+','+file2+','
+		allsents = []
+
+		print 'Open files and find features / supervision tags'
+		for infile in allfiles.split(','):
+			if(len(infile)) :
+				print '-', infile
+				sent = []
+				in_handler = formats.HtmlReader(infile, compatibility_mode=False)
+				for token in in_handler:
+					tag = ''
+					if token.type == 'w' or token.type == 'c':
+						tags = ''
+						if args.pos:
+							for ps in token.gloss.ps:
+								tags += ps
```

Comment:
In multitag ps field the order of PoS tags is not guaranteed, i.e. it is possible to encounter both n/adj and adj/n which refer to the same thing. In your case it will produce two different tags.